### PR TITLE
Add limits on dynamic log message

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Increase the limit on string length in arg template to 8K each for dynamic log only
Put a hard limit on evaluated log message size of 8K. Rename LogTemplateBuilder to StringTemplateBuilder

# Motivation
Allow larger string argument for dynamic log message but with a max to 8K to not breach the 1MB limit for snapshot
# Additional Notes

Jira ticket: [DEBUG-1917]

[DEBUG-1917]: https://datadoghq.atlassian.net/browse/DEBUG-1917?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java [hunk 1]
```diff
@@ -386,8 +388,15 @@ public void evaluate(
       sample(logStatus, methodLocation);
     }
     if (logStatus.isSampled() && logStatus.getCondition()) {
-      LogMessageTemplateBuilder logMessageBuilder = new LogMessageTemplateBuilder(segments);
-      logStatus.setMessage(logMessageBuilder.evaluate(context, logStatus));
+      StringTemplateBuilder logMessageBuilder = new StringTemplateBuilder(segments, LIMITS);
+      String msg = logMessageBuilder.evaluate(context, logStatus);
+      if (msg != null && msg.length() > LOG_MSG_LIMIT) {
+        StringBuilder sb = new StringBuilder(LOG_MSG_LIMIT + 3);
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java:394]: wonder if we should copy LOG_MSG_LIMIT - 3 to keep the log message to be under LOG_MSG_LIMIT.

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/agent/LogProbesInstrumentationTest.java [hunk 2]
```diff
@@ -37,6 +37,8 @@ public class LogProbesInstrumentationTest {
...
...0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000";
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/agent/LogProbesInstrumentationTest.java:41]: 🤣

**Review by shatzi — APPROVED** [review]
nice one

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java (hunk 1 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/probe/LogProbe.java:394]: 8K is arbitrary there is nothing that prevent us to add those 3 more.
this way we have 8K effective

**Pull request merged**